graphene/relay/connection.py

```diff
@@ -1,14 +1,36 @@
 import re
 from collections.abc import Iterable
 from functools import partial
+from typing import Type
 
 from graphql_relay import connection_from_array
 
 from ..types import Boolean, Enum, Int, Interface, List, NonNull, Scalar, String, Union
 from ..types.field import Field
 from ..types.objecttype import ObjectType, ObjectTypeOptions
 from ..utils.thenables import maybe_thenable
-from .node import is_node
+from .node import is_node, AbstractNode
+
+
+def get_edge_class(
+    connection_class: Type["Connection"], _node: Type[AbstractNode], base_name: str
+):
+    edge_class = getattr(connection_class, "Edge", None)
+
+    class EdgeBase:
+        node = Field(_node, description="The item at the end of the edge")
+        cursor = String(required=True, description="A cursor for use in pagination")
+
+    class EdgeMeta:
+        description = f"A Relay edge containing a `{base_name}` and its cursor."
+
+    edge_name = f"{base_name}Edge"
+    if edge_class:
+        edge_bases = [edge_class, EdgeBase, ObjectType]
+    else:
+        edge_bases = [EdgeBase, ObjectType]
```

Comment:
Could you elaborate why you explicitly add `[edge_class, EdgeBase, ObjectType]` or `[EdgeBase, ObjectType]`, respectively as edge_bases? 🙂 Wouldn't it be easier to let `EdgeBase` inherit from `ObjectType` and check that edge_class does the same? 🙂 Or alternatively put `EdgeBase` into global scope and ensure that if `edge_class` is set it inherits from `EdgeBase`? 🙂 Would definitely be a stronger restriction on the user side, but I am not sure if it can yield unintended side effects when adding `EdgeBase` and `ObjectType` as additional bases when the user assumes only `edge_class` is being used 🙂

Reply:
I honestly just kept the old behavior to not induce any breaking changes. However, I agree that putting `EdgeBase` into Global Scope helps here! 
If we require inheritance from `EdgeBase`, or expect that in any way, we would create a breaking change :/

Reply:
I am generally fine with moving this to a separate issue. Alternatively we could do an 
```
edge_bases = [edge_class] if edge_class else [EdgeBase]
if not isinstance(edge_class, ObjectType):
    edge_bases = [*edge_bases, ObjectType]
```
This way we don't assume inheritance any inheritance and would add `ObjectType` if it is not inherited from. Alternatively, we could do this with `EdgeBase` if we actually require it to be in edge_bases, but I think ObjectType should be enough right? 🙂

Reply:
Done 🙂

Reply:
👍